src/history/HistoryArchive.cpp

```diff
@@ -108,8 +108,9 @@ void
 HistoryArchiveState::save(std::string const& outFile) const
 {
     ZoneScoped;
-    std::ofstream out(outFile);
+    std::ofstream out;
     out.exceptions(std::ios::failbit | std::ios::badbit);
```

Comment:
I know this was here before, but we should update TxSimGenerateBucketsWork::onSuccess, which is using this function and can throw. 
* we should guard the call to `save` with try/catch and return WORK_FAILURE if there's a problem
* To make the change above work, it needs to be moved to `onRun` https://github.com/stellar/stellar-core/blob/24d1a612aa0ee99db993ee97ceb1891d9a00e9cd/src/transactions/simulation/TxSimGenerateBucketsWork.cpp#L141

Reply:
As for the other subsystems, from what i see, catchup/publish are already properly using `load`/`save`.

Reply:
updated